### ISSUE TITLE
[analyzer] String -> ConstCharStar generates invalid code 

### ISSUE THREAD

**Issue by haxiomic** [HaxeFoundation/haxe#10112]
On the C++ target, `String` can implicitly or explicitly cast to `cpp.ConstCharStar`, however invalid code is generated since 4.1 when `-D analyzer-optimize` is used (this worked in 4.0)

```haxe
class Main {

	static function main() {
		var x = "hello world";
		var cStr: cpp.ConstCharStar = x; // same result with cpp.ConstCharStar.fromString(x)
		trace(cStr);
	}

}
```

generates
```C++
const char* cStr = ( (const char*)( (HX_("hello world",c4,ee,7b,f8).utf8_str()).get_raw()) );
```
but the correct should be
```C++
const char* cStr = ( (const char*)( HX_("hello world",c4,ee,7b,f8).utf8_str() );
```

where the `.get_raw()` should not be there (and isn't there when `-D analyzer-optimize` isn't used)

Bisected to b8289fe

Repro
[haxe-issue-10112.zip](https://github.com/HaxeFoundation/haxe/files/5961317/haxe-issue-10112.zip)

**Comment by Aidan63** [comment]
I think this is the same as #9733. I managed to fix it for my case in #10038, might be worth giving that a go and see if it works for you as well.

**Comment by haxiomic** [comment]
@Aidan63 you're absolutely right, this is the same issue as #9733 and your PR #10038 resolves :)
closing this as duplicate
